dog.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2021 Siôn le Roux.  All rights reserved.
+// Use of this source code is subject to an MIT-style
+// licence which can be found in the LICENSE file.
+
+package main
+
+import (
+	"image"
+	"math"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/solarlune/resolv"
+)
+
+// Coord is the coordinate of a path point
+type Coord struct {
+	X float64
+	Y float64
+}
+
+// Path is an array of coordinates
+type Path []Coord
+
+// dogSpeed is the distance the dog moves per update cycle
+const dogSpeed float64 = 1
+
+// waitingRadius is the maximum distance the dog walks away from the player
+const waitingRadius float64 = 96
+
+// zombieDangerRadius: if a zombie is this close to the dog, it runs away
+const zombieDangerRadius float64 = 128
+
+// zombieSafeRadius: if a zombie is at least this far from the dog, it stops running
+const zombieSafeRadius float64 = 192
+
+// states of the dog
+// It would be great to map them to the frameTag.Name from JSON
+const (
+	dogWalking  int = 0
+	dogSniffing     = 1
+	dogSitting      = 2
```

Comment:
If you have a sequence of numbers you're using as constants, then instead of writing them out one-by-one like this you can use iota instead, [like in this game for example](https://github.com/sinisterstuf/cr1ckt/blob/master/internal/game.go#L497-L507), so you can just drop off the numbers and it will work:

```suggestion
	dogWalking  int = iota
	dogSniffing
	dogSitting
```

and because it's an int by default you can leave that off too. 😉